pkg/cluster/spec/grafana.go

```diff
@@ -273,14 +278,54 @@ func (i *GrafanaInstance) InitConfig(
 	if len(monitors) == 0 {
 		return errors.New("no prometheus found in topology")
 	}
+
+	// Create datasources configuration
+	datasources := make([]*config.DatasourceConfig, 0)
+
+	// Add default Prometheus datasource
+	defaultDatasource := config.NewDatasourceConfig(
+		clusterName,
+		fmt.Sprintf("http://%s", utils.JoinHostPort(monitors[0].Host, monitors[0].Port)),
+	)
+	datasources = append(datasources, defaultDatasource)
+
+	// Add VM datasource if enabled
+	if monitors[0].EnableVMRemoteWrite {
+		vmDatasource := config.NewDatasourceConfig(
+			fmt.Sprintf("%s-vm", clusterName),
+			fmt.Sprintf("http://%s", utils.JoinHostPort(monitors[0].Host, monitors[0].NgPort)),
```

Comment:
What If TLS enabled?

Reply:
I tested enabling TLS, but the monitoring service isn't using it. It seems TLS isn’t enabled for monitoring.

Reply:
ptal @nexustar